pkg/instrumentors/bpf/database/sql/probe.go

```diff
@@ -0,0 +1,239 @@
+		return err
+	}
+
+	up, err := ctx.Executable.Uprobe("", h.bpfObjects.UprobeQueryDC, &link.UprobeOptions{
+		Address: offset,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	h.uprobes = append(h.uprobes, up)
+
+	retOffsets, err := ctx.TargetDetails.GetFunctionReturns(h.FuncNames()[0])
+
+	if err != nil {
+		return err
+	}
+
+	for _, ret := range retOffsets {
+		retProbe, err := ctx.Executable.Uprobe("", h.bpfObjects.UuprobeQueryDC_Returns, &link.UprobeOptions{
+			Address: ret,
+		})
+		if err != nil {
+			return err
+		}
+		h.returnProbs = append(h.returnProbs, retProbe)
+	}
+
+	rd, err := perf.NewReader(h.bpfObjects.Events, os.Getpagesize())
+	if err != nil {
+		return err
+	}
+	h.eventsReader = rd
+
+	return nil
+}
+
+// Run runs the events processing loop.
+func (h *Instrumentor) Run(eventsChan chan<- *events.Event) {
+	logger := log.Logger.WithName("database/sql/sql-instrumentor")
+	var event Event
+	for {
+		record, err := h.eventsReader.Read()
+		if err != nil {
+			if errors.Is(err, perf.ErrClosed) {
+				return
+			}
+			logger.Error(err, "error reading from perf reader")
+			continue
+		}
+
+		if record.LostSamples != 0 {
+			logger.V(0).Info("perf event ring buffer full", "dropped", record.LostSamples)
+			continue
+		}
+
+		if err := binary.Read(bytes.NewBuffer(record.RawSample), binary.LittleEndian, &event); err != nil {
+			logger.Error(err, "error parsing perf event")
+			continue
+		}
+
+		eventsChan <- h.convertEvent(&event)
+	}
+}
+
+func (h *Instrumentor) convertEvent(e *Event) *events.Event {
+	query := unix.ByteSliceToString(e.Query[:])
+
+	sc := trace.NewSpanContext(trace.SpanContextConfig{
+		TraceID:    e.SpanContext.TraceID,
+		SpanID:     e.SpanContext.SpanID,
+		TraceFlags: trace.FlagsSampled,
+	})
+
+	var pscPtr *trace.SpanContext
+	if e.ParentSpanContext.TraceID.IsValid() {
+		psc := trace.NewSpanContext(trace.SpanContextConfig{
+			TraceID:    e.ParentSpanContext.TraceID,
+			SpanID:     e.ParentSpanContext.SpanID,
+			TraceFlags: trace.FlagsSampled,
+			Remote:     true,
+		})
+		pscPtr = &psc
+	} else {
+		pscPtr = nil
+	}
+
+	return &events.Event{
+		Library:     h.LibraryName(),
+		Name:        "DB",
+		Kind:        trace.SpanKindClient,
+		StartTime:   int64(e.StartTime),
+		EndTime:     int64(e.EndTime),
+		SpanContext: &sc,
+		Attributes: []attribute.KeyValue{
+			semconv.DBStatementKey.String(query),
+		},
+		ParentSpanContext: pscPtr,
+	}
+}
+
+// Close stops the Instrumentor.
+func (h *Instrumentor) Close() {
+	log.Logger.V(0).Info("closing database/sql/sql instrumentor")
+	if h.eventsReader != nil {
+		h.eventsReader.Close()
+	}
+
+	for _, r := range h.uprobes {
+		r.Close()
+	}
+
+	for _, r := range h.returnProbs {
+		r.Close()
+	}
+
+	if h.bpfObjects != nil {
+		h.bpfObjects.Close()
+	}
+}
+
+// shouldIncludeDBStatement returns if the user has configured SQL queries to be included.
+func shouldIncludeDBStatement() bool {
+	val, exists := os.LookupEnv(IncludeDBStatementEnvVar)
+	if exists {
```

Comment:
```suggestion
	val := os.Getenv(IncludeDBStatementEnvVar)
	if val != "" {
```

Reason:

> The SDK MUST interpret an empty value of an environment variable the same way as when the variable is unset.

From https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#parsing-empty-value